course_discovery/apps/course_metadata/data_loaders/geolocation_loader.py

```
@@ -0,0 +1,210 @@
+
+        # check the values for each field
+        is_valid_uuid = self.is_valid_uuid(data['uuid'])
+        if not is_valid_uuid:
+            error_msgs.append('Invalid UUID: {}'.format(data['uuid']))
+
+        is_course_or_program = data['product_type'] == 'course' or data['product_type'] == 'program'
+        if not is_course_or_program:
+            error_msgs.append('Wrong product_type value for UUID: {}'.format(data['uuid']))
+
+        if is_valid_uuid:  # only check if item exists if the provided uuid is valid
+            item_found = self.validate_course_or_program(data)
+            if not item_found:
+                error_msgs.append('Course or Program with UUID {} was not found'.format(data['uuid']))
+
+        if error_msgs:
+            return ', '.join(error_msgs)
+        
+        return ''
+
+    def is_valid_uuid(self, val):
+        try:
+            uuid.UUID(str(val))
+            return True
+        except ValueError:
+            return False
+ 
+    def validate_course_or_program(self, data):
+        """
+        Verify if the given course or program exists
+        Returns False if product could not be found
+        """
+        model = None
+        if data['product_type'] == 'course':
+            model = Course
+        elif data['product_type'] == 'program':
+            model = Program
+
+        if not model:
+            return False
+
+        return self.validate_product_exists(model, 'uuid', data['uuid'])
+
+    def validate_product_exists(self, model, key, value):
+        """
+        Get an object from the database by its key and value
+        """
+        model_name = model._meta.object_name
+        kwrags = {key: value}
+        try:
+            return (
+                model.everything.filter(**kwrags).exists() if model_name == 'Course'
+                else model.objects.filter(**kwrags).exists()
+            )
+        except model.DoesNotExist:
```

filter() does not raise this exception. The return of filter is QuerySet and it is empty if the objects are not located.

---

makes sense, do you think this should be converted into a catch-all exception handler? Or should we remove the try/except block all together?

---

try/except can be removed. The queryset can be converted into a list and then any existence checks can be performed on the list.